### PR TITLE
Add splitter support and add icons to buttons

### PR DESCRIPTION
**Pull request by jacob314**
The slider functionality is mainly for the inspector but is convenient for the debugger as well.
In a future CL we should polish how the debugger windows look when narrow. See the attached animation of the slider.
The slider js library used only adds about 2kb to the app size.

screen shot 2019-01-05 at 1 36 07 pm

slider_example

### CODE REVIEW

**Comment by devoncarew** [comment]
I see dart2js having an issue w/ this PR: https://travis-ci.org/flutter/devtools/jobs/475814926#L311

**Comment by devoncarew** [comment]
and, 👍 for having a splitter - it'll make it easier for people to use the UI

File: lib/ui/primer.dart [hunk 1]
```diff
@@ -5,6 +5,9 @@
 import 'dart:async';
 import 'dart:html';
 
+import 'package:devtools/ui/html_icon_renderer.dart';
```
[the remark this answers is not in the document]
Comment (jacob314) [reply lib/ui/primer.dart]: done. I wish the intellij quickfix used relative imports when possible.

File: lib/ui/split.dart [hunk 2]
```diff
@@ -0,0 +1,123 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
```
[the remark this answers is not in the document]
Comment (jacob314) [reply lib/ui/split.dart]: done

File: lib/ui/split.dart [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+// found in the LICENSE file.
+
+/// Thin JS interop wrapper around https://split.js.org/
+@JS()
+library splitter;
+
+import 'dart:async';
+import 'dart:html';
+
+import 'package:devtools/ui/elements.dart';
```
[the remark this answers is not in the document]
Comment (jacob314) [reply lib/ui/split.dart]: done

File: lib/ui/primer.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (devoncarew) [reply lib/ui/primer.dart]: Me as well. It should default to relative imports, and only use package: self references on some user setting, or if it detects that pattern in the current library.

File: lib/ui/ui_utils.dart [hunk 4]
```diff
@@ -4,14 +4,21 @@
 
 import 'dart:html' as html;
 
+import 'package:devtools/service_extensions.dart';
```
[the remark this answers is not in the document]
Comment (jacob314) [reply lib/ui/ui_utils.dart]: done

File: lib/ui/ui_utils.dart [hunk 5]
```diff
@@ -32,40 +39,36 @@ PButton createExtensionButton(String text, String extensionName) {
...
-  // Disable checkbox for unavailable service extensions.
-  if (!serviceManager.serviceExtensionManager
-      .isServiceExtensionAvailable(extensionName)) {
-    input.disabled = true;
-    text.element.style.color = disabledTextColor;
-  }
-  serviceManager.serviceExtensionManager.hasServiceExtension(extensionName,
-      (available) {
-    input.disabled = !available;
-    text.element.style.color = available ? 'black' : disabledTextColor;
```
Comment (kenzieschmoll) [inline lib/ui/ui_utils.dart]: Why did we get rid of the disabled text color?

File: web/index.html [hunk 6]
```diff
@@ -23,6 +23,8 @@
     <link rel="stylesheet" href="packages/devtools/logging/logging.css">
 
     <script type="text/javascript" src="https://www.gstatic.com/charts/loader.js"></script>
+    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/split.js/1.5.9/split.min.js"></script>
```
[the remark this answers is not in the document]
Comment (jacob314) [reply web/index.html]: Sg. added a third_party/split directory under web.

**Comment by devoncarew** [comment]
Let's add a license file for it as well, similar to https://github.com/flutter/flutter-intellij/tree/master/third_party/vmServiceDrivers

File: lib/ui/ui_utils.dart (hunk 5 above)
Comment (jacob314) [reply lib/ui/ui_utils.dart]: good catch. I lost this on a merge.
I've added back this functionality by tweaking the HTML slightly so all the label content is within a label tag and the CSS controls fading out the label and icon when disabled.

**Comment by jacob314** [comment]
Done. Added a LICENSE file with the license copied out of readme.md

**Pull request merged**